[RNMobile] Fix Android crash when closing the editor while dragging

fluiddot opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Bumps the version of `react-native-reanimated` dependency to include [this fix](https://github.com/wordpress-mobile/react-native-reanimated/pull/15).

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Addresses the crash outlined in https://github.com/wordpress-mobile/gutenberg-mobile/issues/4801.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

This PR basically changes the version of `react-native-reanimated` dependency. Additionally, `react-native-reanimated` dependency has been excluded from `react-native-gesture-handler` as it's already included. This way we remove the transitive dependency and prevent conflicts due to using different versions of the same dependency.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
1. Open the app and connect it to the Metro server.
1. Start dragging a block.
1. While moving the block around the screen, tap the back button with another finger to close the editor.
1. Observe that the app doesn't crash.

## Screenshots or screencast <!-- if applicable -->
N/A

fluiddot committed:
Remove Reanimated transitive dependency on Android

fluiddot committed:
Test Android crash fix by changing Reanimated version

fluiddot committed:
Bump react-native-reanimated dependency version

fluiddot committed:
Bump react-native-reanimated dependency on Android

fluiddot commented on packages/react-native-bridge/android/react-native-bridge/build.gradle:
```diff
@@ -70,13 +70,18 @@ dependencies {
     implementation "org.wordpress-mobile:react-native-webview:${extractPackageVersion(packageJson, 'react-native-webview', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-clipboard:${extractPackageVersion(packageJson, '@react-native-clipboard/clipboard', 'dependencies')}"
+
+    implementation("com.github.wordpress-mobile:react-native-gesture-handler:${extractPackageVersion(packageJson, 'react-native-gesture-handler', 'dependencies')}", {
+        // Remove Reanimated transitive dependency as it's already defined here
+        exclude group: 'com.github.wordpress-mobile', module: 'react-native-reanimated'
+    })
```
The transitive dependency is defined [here](https://github.com/wordpress-mobile/react-native-gesture-handler/blob/1a4713873e9aeba0616f3f1e180595e6101cd3bd/android/build.gradle#L94). Once https://github.com/wordpress-mobile/react-native-reanimated/pull/15 is merged, we could update it so the same version is referenced in both places. I'd like to note that this is not mandatory as it won't affect the editor, however, it's recommended to prevent build issues in `react-native-gesture-handler` due to using different versions.

fluiddot commented on packages/react-native-bridge/android/react-native-bridge/build.gradle:
```diff
@@ -70,13 +70,18 @@ dependencies {
     implementation "org.wordpress-mobile:react-native-screens:${extractPackageVersion(packageJson, 'react-native-screens', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-safe-area-context:${extractPackageVersion(packageJson, 'react-native-safe-area-context', 'dependencies')}"
-    implementation "com.github.wordpress-mobile:react-native-reanimated:${extractPackageVersion(packageJson, 'react-native-reanimated', 'dependencies')}"
+    implementation 'com.github.wordpress-mobile:react-native-reanimated:9e63336aa24daeefb9f756bbf8cdddfa2f3d44f5'
```
This change will be reverted once https://github.com/wordpress-mobile/react-native-reanimated/pull/15 is merged into the `wp-fork` branch, and we create a new release with tag `2.4.1-wp-2`.

fluiddot commented on packages/react-native-editor/package.json:
```diff
@@ -65,7 +65,7 @@
 		"react-native-modal": "^11.10.0",
 		"react-native-prompt-android": "https://raw.githubusercontent.com/wordpress-mobile/react-native-prompt-android/v1.0.0-wp-2/react-native-prompt-android-1.0.0-wp-2.tgz",
-		"react-native-reanimated": "https://raw.githubusercontent.com/wordpress-mobile/react-native-reanimated/2.4.1-wp-1/react-native-reanimated-2.4.1-wp-1.tgz",
+		"react-native-reanimated": "https://raw.githubusercontent.com/wordpress-mobile/react-native-reanimated/fix-android-layout-reanimation-crash/react-native-reanimated-2.4.1-wp-2.tgz",
```
Similar to [this comment](https://github.com/WordPress/gutenberg/pull/40810/files#r864651892), this change will be updated once https://github.com/wordpress-mobile/react-native-reanimated/pull/15 is merged. The dependency should point to:
```
https://raw.githubusercontent.com/wordpress-mobile/react-native-reanimated/2.4.1-wp-2/react-native-reanimated-2.4.1-wp-2.tgz
```

fluiddot committed:
Update package-lock.json file

fluiddot commented:
@geriux once https://github.com/wordpress-mobile/react-native-reanimated/pull/15 is approved and merged, I'll update the dependency references in this PR and set it ready for review. In the meantime, feel free to take a look and provide any feedback, thanks 🙇 .

geriux commented:
> @geriux once [wordpress-mobile/react-native-reanimated#15](https://github.com/wordpress-mobile/react-native-reanimated/pull/15) is approved and merged, I'll update the dependency references in this PR and set it ready for review. In the meantime, feel free to take a look and provide any feedback, thanks 🙇 .

I tested these changes and its working well! 🎉  The crash is no longer present, so once this is ready for review and updated with the references I'll give it the ✅

fluiddot committed:
Bump react-native-reanimated dependency version

fluiddot committed:
Revert "Test Android crash fix by changing Reanimated version"

This reverts commit d01cdae6ce09e98a4bf42ab11719fbe2b753b288.

fluiddot commented:
> > @geriux once [wordpress-mobile/react-native-reanimated#15](https://github.com/wordpress-mobile/react-native-reanimated/pull/15) is approved and merged, I'll update the dependency references in this PR and set it ready for review. In the meantime, feel free to take a look and provide any feedback, thanks 🙇 .
> 
> I tested these changes and its working well! 🎉 The crash is no longer present, so once this is ready for review and updated with the references I'll give it the ✅

Thanks @geriux for testing the PR 🙇 ! I've just merged https://github.com/wordpress-mobile/react-native-reanimated/pull/15 and updated the references, so the PR is ready for review.

geriux reviewed (approved):
LGTM! I tested this again using a Redmi Note 8T to double-check and it's working as expected 🚀
